src/random.cr

```diff
@@ -277,6 +290,115 @@ module Random
+  # ```
+  def random_bytes(buf : Bytes) : Nil
+    n = buf.size / sizeof(typeof(next_u))
+    remaining = buf.size - n * sizeof(typeof(next_u))
+
+    slice = buf.to_unsafe.as(typeof(next_u)*).to_slice(n)
+    slice.each_index { |i| slice[i] = next_u }
+
+    if remaining > 0
+      bytes = next_u
+      remaining.times do |i|
+        bits = i * 8
+        mask = typeof(next_u).new(0xff << bits)
+        buf[-i - 1] = UInt8.new((bytes & mask) >> bits)
+      end
+    end
+  end
+
+  # Generates a slice filled with *n* random bytes.
+  #
+  # ```
+  # Random.new.random_bytes    # => [145, 255, 191, 133, 132, 139, 53, 136, 93, 238, 2, 37, 138, 244, 3, 216]
+  # Random.new.random_bytes(4) # => [217, 118, 38, 196]
+  # ```
+  def random_bytes(n : Int = 16) : Bytes
+    raise ArgumentError.new "Negative size: #{n}" if n < 0
+    Bytes.new(n).tap { |buf| random_bytes(buf) }
+  end
+
+  # Generates *n* random bytes that are encoded into base64.
+  #
+  # Check `Base64#strict_encode` for details.
+  #
+  # ```
+  # Random::System.base64(4) # => "fK1eYg=="
+  # ```
+  #
+  # It is recommended to use the secure `Random::System` as a source or another
+  # cryptographically quality PRNG such as `Random::ISAAC` or ChaCha20.
+  def base64(n : Int = 16) : String
+    Base64.strict_encode(random_bytes(n))
+  end
+
+  # URL-safe variant of `#base64`.
+  #
+  # Check `Base64#urlsafe_encode` for details.
+  #
+  # ```
+  # Random::System.urlsafe_base64                    # => "MAD2bw8QaBdvITCveBNCrw"
+  # Random::System.urlsafe_base64(8, padding: true)  # => "vvP1kcs841I="
+  # Random::System.urlsafe_base64(16, padding: true) # => "og2aJrELDZWSdJfVGkxNKw=="
+  # ```
+  #
+  # It is recommended to use the secure `Random::System` as a source or another
+  # cryptographically quality PRNG such as `Random::ISAAC` or ChaCha20.
+  def urlsafe_base64(n : Int = 16, padding = false) : String
+    Base64.urlsafe_encode(random_bytes(n), padding)
+  end
+
+  # Generates a hexadecimal string based on *n* random bytes.
+  #
+  # The bytes are encoded into a string of two-digit hexadecimal
+  # number (00-ff) per byte.
+  #
+  # ```
+  # Random::System.hex    # => "05f100a1123f6bdbb427698ab664ff5f"
+  # Random::System.hex(1) # => "1a"
+  # ```
+  #
+  # It is recommended to use the secure `Random::System` as a source or another
+  # cryptographically quality PRNG such as `Random::ISAAC` or ChaCha20.
+  def hex(n : Int = 16) : String
+    random_bytes(n).hexstring
+  end
+
+  # Generates a UUID (Universally Unique Identifier).
+  #
+  # It generates a random v4 UUID. See
+  # [RFC 4122 Section 4.4](https://tools.ietf.org/html/rfc4122#section-4.4)
+  # for the used algorithm and its implications.
+  #
+  # ```
+  # Random::System.uuid # => "a4e319dd-a778-4a51-804e-66a07bc63358"
+  # ```
+  #
+  # It is recommended to use the secure `Random::System` as a source or another
+  # cryptographically quality PRNG such as `Random::ISAAC` or ChaCha20.
+  def uuid : String
```

Comment:
I'd like UUID to be moved into it's own struct with methods before 1.0 so hopefully this method should disappear and be replaced by a constructor on that. Doesn't help with the hex/base64 methods though. I don't think that using `Random::DEFAULT.hex` will be a world-ending issue in Crystal anyway (not that it's usually the correct thing to do) because PCG32 has fairly strong guarantees.